rosetta/transactor/transactor.go

```diff
@@ -0,0 +1,592 @@
+		}
+	}
+
+	// Verify the transaction script is the token transfer script.
+	script, err := t.generate.TransferTokens(dps.FlowSymbol)
+	if err != nil {
+		return identifier.Block{}, 0, nil, nil, fmt.Errorf("could not generate transfer script: %w", err)
+	}
+	if !bytes.Equal(script, tx.Script) {
+		return identifier.Block{}, 0, nil, nil, failure.InvalidScript{
+			Script:      string(tx.Script),
+			Description: failure.NewDescription("transaction text is not valid token transfer script"),
+		}
+	}
+
+	// Verify that the transaction script has the correct number of arguments.
+	args := tx.Arguments
+	if len(args) != argsRequired {
+		return identifier.Block{}, 0, nil, nil, failure.InvalidArguments{
+			Have:        uint(len(args)),
+			Want:        argsRequired,
+			Description: failure.NewDescription("invalid number of arguments"),
+		}
+	}
+
+	// Parse and validate the amount argument.
+	val, err := cjson.Decode(args[0])
+	if err != nil {
+		return identifier.Block{}, 0, nil, nil, failure.InvalidAmount{
+			Amount: string(args[0]),
+			Description: failure.NewDescription("could not parse transaction amount",
+				failure.WithErr(err)),
+		}
+	}
+	amountArg, ok := val.ToGoValue().(uint64)
+	if !ok {
+		return identifier.Block{}, 0, nil, nil, failure.InvalidAmount{
+			Amount:      string(args[0]),
+			Description: failure.NewDescription("invalid amount"),
+		}
+	}
+	amount := strconv.FormatUint(amountArg, 10)
+
+	// Parse and validate receiver script argument.
+	val, err = cjson.Decode(args[1])
+	if err != nil {
+		return identifier.Block{}, 0, nil, nil, failure.InvalidReceiver{
+			Receiver: string(args[1]),
+			Description: failure.NewDescription("could not parse transaction receiver address",
+				failure.WithErr(err)),
+		}
+	}
+	addr := flow.HexToAddress(val.String())
+	receiver := identifier.Account{
+		Address: addr.String(),
+	}
+	_, err = t.validate.Account(receiver)
+	if err != nil {
+		return identifier.Block{}, 0, nil, nil, fmt.Errorf("invalid receiver account: %w", err)
+	}
+
+	// Validate the reference block identifier.
+	refBlockID := identifier.Block{
+		Hash: tx.ReferenceBlockID.String(),
+	}
+	height, blockID, err := t.validate.Block(refBlockID)
+	if err != nil {
+		return identifier.Block{}, 0, nil, nil, fmt.Errorf("invalid reference block: %w", err)
+	}
+
+	// Create the send operation.
+	sendOp := object.Operation{
+		ID: identifier.Operation{
+			Index: 0,
+		},
+		AccountID: sender,
+		Type:      dps.OperationTransfer,
+		Amount: object.Amount{
+			Value: "-" + amount,
+			Currency: identifier.Currency{
+				Symbol:   dps.FlowSymbol,
+				Decimals: dps.FlowDecimals,
+			},
+		},
+	}
+
+	// Create the receive operation.
+	receiveOp := object.Operation{
+		ID: identifier.Operation{
+			Index: 1,
+		},
+		AccountID: receiver,
+		Type:      dps.OperationTransfer,
+		Amount: object.Amount{
+			Value: amount,
+			Currency: identifier.Currency{
+				Symbol:   dps.FlowSymbol,
+				Decimals: dps.FlowDecimals,
+			},
+		},
+	}
```

Comment:
Why don't we set the transaction status? Does it not matter? Do we not know it?

Reply:
Same question for the `NetworkIndex` attribute.

Reply:
The operations here correspond to events that might not even have happened on the chain yet, as the `Parse` endpoint is used for both unsigned and signed transactions, that may or may not have been submitted yet.

- the transaction status _MUST_ not be set if the transaction has not yet been submitted; this is hard to determine in the Flow system;
- the network index is optional and only makes sense after the transaction has been finalized/executed; but it would be complex to infer.

What do you think @Maelkum?

Reply:
For the status field I think we should definitely omit it.

For the network index I don't know - it was mentioned that it should be populated if possible. But I'm not sure how useful it is. Ethereum implementation also does not populate this.

I'm leaning also towards omitting it now. But we should be careful to not omit it where it's relevant, which is in the data API - there the value of 0 can be valid and should be present, right?

Perhaps we should do what we did for the `height` parameter and make it a pointer, so there's a difference between unset (nil) and set but zero.